theme/templates/base.html

```diff
@@ -49,6 +52,31 @@
         </nav>
         <!-- content -->
         {% block content %}{% endblock %}
+        <!-- Newsletter Form Section -->
+        <section class="newsletter secondary">
```

Comment:
I think this section should be `col-12` as well and then the styles should adjust the necessary padding/margin, etc.
I think on all screens the darker gray box should have the paddings as the rest of the page and then the text field and button should resize.

You can also use `pelican -r ./content -o ./output -s ./publishconf.py` to preview the production site locally (will load the disqus comments form on the blog so you can compare).